modules/publicize/ui.php

```diff
@@ -584,180 +584,87 @@ function post_page_metabox() {
-					}
-
-					// Determine the state of the checkbox (on/off) and allow filtering
-					$checked = $skip != 1 || $done;
-					/**
-					 * Filter the checkbox state of each Publicize connection appearing in the post editor.
-					 *
-					 * @module publicize
-					 *
-					 * @since 2.0.1
-					 *
-					 * @param bool $checked Should the Publicize checkbox be enabled for a given service.
-					 * @param int $post->ID Post ID.
-					 * @param string $name Service name.
-					 * @param array $connection Array of connection details.
-					 */
-					$checked = apply_filters( 'publicize_checkbox_default', $checked, $post->ID, $name, $connection );
-
-					// Force the checkbox to be checked if the post was DONE, regardless of what the filter does
-					if ( $done ) {
-						$checked = true;
-					}
-
-					// This post has been handled, so disable everything
-					if ( $all_done ) {
-						$disabled = ' disabled="disabled"';
-					}
-
-					$label = sprintf(
-						_x( '%1$s: %2$s', 'Service: Account connected as', 'jetpack' ),
-						esc_html( $this->publicize->get_service_label( $name ) ),
-						esc_html( $this->publicize->get_display_name( $name, $connection ) )
-					);
-					if ( !$skip || $done ) {
-						$active[] = $label;
-					}
-					?>
-					<li>
-						<label for="wpas-submit-<?php echo esc_attr( $unique_id ); ?>">
-							<input type="checkbox" name="wpas[submit][<?php echo $unique_id; ?>]" id="wpas-submit-<?php echo $unique_id; ?>" class="wpas-submit-<?php echo $name; ?>" value="1" <?php
-								checked( true, $checked );
-								echo $disabled;
-							?> />
-							<?php
-							if ( $hidden_checkbox ) {
-								// Need to submit a value to force a global connection to post
-								echo '<input type="hidden" name="wpas[submit][' . $unique_id . ']" value="1" />';
-							}
-							echo esc_html( $label );
-							?>
-						</label>
-					</li>
-					<?php
-				}
-			}
+		$all_done = $this->publicize->done_sharing_post();
+		$connection_list = $this->publicize->get_filtered_connection_data();
 
-			if ( $title = get_post_meta( $post->ID, $this->publicize->POST_MESS, true ) ) {
-				$title = esc_html( $title );
-			} else {
-				$title = '';
+		ob_start();
+		?>
+			<div id="publicize-form" class="hide-if-js">
+				<ul>
+		<?php
+
+		foreach ( $connection_list as $c ) {
+			if ( $c['active'] ) {
+				$active_list[] = $c['label'];
 			}
 			?>
+				<li>
+					<label for="wpas-submit-<?php echo esc_attr( $c['unique_id'] ); ?>">
+						<input type="checkbox" name="wpas[submit][<?php echo $c['unique_id']; ?>]" id="wpas-submit-<?php echo $c['unique_id']; ?>" class="wpas-submit-<?php echo $c['name']; ?>" value="1" <?php
+						checked( true, $c['checked'] );
+						if ( $c['disabled'] ) {
+							echo 'disabled="disabled"';
+						}
+						?> />
+						<?php
+						if ( $c['hidden_checkbox'] ) {
+							// Need to submit a value to force a global connection to post
```

Comment:
This logic doesn't change in this PR, but we should check and see if it's this hidden form field that controls the server's behavior regarding global connections. If that's true, it's strange.

Reply:
I agree. If we're able to ditch the hidden field, I guess we should also be able to remove [the field](https://github.com/Automattic/jetpack/pull/9955#discussion_r227637118) from `get_filtered_connection_data()`'s return value, and filter the corresponding connection from that array instead?